src/cli/api/local_api.py

```diff
@@ -382,20 +526,266 @@ def get_results(self, path: Path, all_results: bool) -> List[ResultType]:
+            for role in application_network_data['roles']:
+                roles.append(role.lower())
+
+            # Local validation
+            if local:
+                if not experiment_input_path.exists():
+                    error_dict['warning'].append(f"Directory not found. {experiment_input_path} does not exist")
+
+                # TODO: Update Confluence with json files instead of yaml files (network.json, application.json and
+                #  result.json
+                # network.json is checked in an earlier validation, so no need to check again
+                if experiment_input_path.exists():
+                    if not (experiment_input_path / 'application.json').exists() \
+                     or not (experiment_input_path / 'result.json').exists():
+                        error_dict['warning'].append(f"{experiment_input_path} needs to contain the files "
+                                                     f"'network.json', 'application.json' and 'result.json'")
+
+                    # Check if the roles from network.json exist in the experiment/input directory
+                    missing_files = []
+                    for item in ['app_' + s + '.py' for s in roles]:
+                        if not (experiment_input_path / item).is_file():
+                            missing_files.append(item)
+
+                    if missing_files:
+                        error_dict['warning'].append(f"{experiment_input_path} is missing the files: {missing_files}")
+
+    # TODO:
+    #  - Update confluence sequence diagram with new function
+    #  - Roundsetmanager validate_asset() should be used in the future when multiple rounds are used.
+    def validate_experiment_json_files(self, path: Path, error_dict: ErrorDictType) -> None:
+        """
+        This function validates if experiment.json contains valid json and if it passes schema validation.
+
+        Args:
+            path: The location of the experiment
+            error_dict: Dictionary containing error and warning messages of the validations that failed
+        """
+
+        experiment_file_path = path / 'experiment.json'
+        experiment_input_path = path / 'input'
+
+        # Validate experiment.json
+        if os.path.isfile(experiment_file_path):
+            experiment_schema = Path(os.path.join(BASE_DIR, 'schema', 'experiments', 'experiment.json', ''))
+
+            json_valid, message = validate_json_file(experiment_file_path)
+            if json_valid:
+                schema_valid, ve = validate_json_schema(experiment_file_path, experiment_schema)
+                if not schema_valid:
+                    error_dict['error'].append(ve)
+
+                experiment_data = read_json_file(experiment_file_path)
+                experiment_network_slug = experiment_data['asset']['network']['slug']
+
+                # Check if the chosen network exists
+                if self._get_network_info(experiment_network_slug):
```

Comment:
Not wrong but return value None of _get_network_info is checked with 'is not None'.